src/Microsoft.AspNetCore.SignalR.Client/InvocationRequest.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    internal class InvocationRequest : IDisposable
+    {
+        private readonly TaskCompletionSource<object> _completionSource;
+        private readonly CancellationTokenRegistration _cancellationTokenRegistration;
+        private readonly bool _streaming;
+        private readonly ILogger _logger;
+        private readonly InvocationSubject _subject;
+
+        public Type ResultType { get; }
+        public CancellationToken CancellationToken { get; }
+        public string InvocationId { get; }
+
+        public Task<object> Task => _completionSource?.Task;
```

Comment:
A null Task is pretty crappy. Make it return a completed task if there's nothing.